app/src/main/java/com/example/taskyy/ui/screens/ReminderScreen.kt

```diff
@@ -71,60 +83,113 @@ fun Body(itemBoxTitle: String, reminderText: String, dateString: String) {
         ItemBoxTitle("Reminder")
     }
     Column {
-        NewItemRow()
-        RowDivider(30)
-        ItemDescription()
-        RowDivider(25)
-        TimeAndDateRow(dateString = dateString)
-        RowDivider(topPadding = 25)
-        Alarm()
+        NewItemRow(title = state.reminderTitleText)
+        RowDivider()
+        ItemDescription(description = state.reminderDescription)
+        RowDivider()
+        TimeAndDateRow(
+            dateString = dateString,
+            onEvent = onEvent,
+            isDatePickerExpanded = state.isDatePickerExpanded,
+            selectedTime = state.selectedTime,
+            selectedDate = state.selectedDate
+        )
+        RowDivider()
+        SetAlarmTimeRow(
+            isAlarmSelectionExpanded = state.isAlarmSelectionExpanded,
+            alarmTimeSelectionText = state.alarmReminderTimeSelection,
+            onEvent = { alarmSectionExpanded ->
+                onEvent(
+                    ReminderEvent.AlarmTypeDropDownSelected(alarmSectionExpanded)
+                )
+            })
     }
 }
 
 @Composable
-fun Alarm() {
+fun SetAlarmTimeRow(
+    isAlarmSelectionExpanded: Boolean,
+    alarmTimeSelectionText: String,
+    onEvent: (Boolean) -> Unit
```

Comment:
For any lambda that does not pass up a specific screen event, I'd use more descriptive names of what they actually handle. In this case, something like `onToggleVisibility: (Boolean) -> Unit` is way more readable